sa/storage-authority.go

```diff
@@ -611,18 +623,64 @@ func (ssa *SQLStorageAuthority) FinalizeAuthorization(authz core.Authorization)
 	return
 }
 
+// NewCertificateRequest creates a new certificate request in the database,
+// assigning it a random ID.  The caller is responsible for setting all fields
+// besides ID.  No fields besides "Status" can be updated.
+func (ssa *SQLStorageAuthority) NewCertificateRequest(req core.CertificateRequest) (output core.CertificateRequest, err error) {
+	if !req.ReadyForSA() {
+		err = fmt.Errorf("Incomplete certificate request sent for storage")
+		return
+	}
+
+	tx, err := ssa.dbMap.Begin()
+	if err != nil {
+		return
+	}
+
+	// Check that it doesn't exist already
+	req.ID = core.NewToken()
+	for existingCertificateRequest(tx, req.ID) {
+		req.ID = core.NewToken()
+	}
+
+	// Insert the request into the database
+	err = tx.Insert(&req)
+	if err != nil {
+		tx.Rollback()
+		return
+	}
+
+	err = tx.Commit()
+	output = req
+	return
+}
+
+// UpdateCertificateRequestStatus sets the "status" field of the
+// identified CertificateRequest to the specified value.
+func (ssa *SQLStorageAuthority) UpdateCertificateRequestStatus(id string, status core.AcmeStatus) (err error) {
+	_, err = ssa.dbMap.Exec("UPDATE `certificateRequests` SET status = ? WHERE id = ?", string(status), id)
```

Comment:
So your previous query used the named interface rather than the anonymous one. If you feel strongly about it perhaps we should set a standard.